src/framework/theme/components/icon/icon.component.ts

```diff
@@ -218,14 +218,17 @@ export class NbIconComponent implements NbIconConfig, OnChanges, OnInit {
   }
 
   ngOnChanges() {
-    if (this.iconDef) {
-      this.iconDef = this.renderIcon(this.icon, this.pack, this.options);
-    }
+    this.iconDef = this.renderIcon(this.icon, this.pack, this.options);
   }
 
   renderIcon(name: string, pack?: string, options?: { [name: string]: any }) {
     const iconDefinition = this.iconLibrary.getIcon(name, pack);
 
+    if (!iconDefinition) {
+      this.clearIcon();
+      return;
+    }
```

Comment:
I found a bit confusing that calling `renderIcon` could clear it instead. Am I missing a case where we need such logic? Could we check if icon def exists before calling render and if it exists call render, otherwise call clear? Something like:
```
ngOnChanges() {
  const iconDef = this.iconLibrary.getIcon(name, pack);
  if (iconDef) {
	this.render(...);
  } else {
    this.clear();
  }
}
```

Anyway, if we keep new logic inside render, please add return types for `renderIcon` methods.

Reply:
Refactored in d79b2dc